### DISCUSSION TITLE
Change name to aiomqtt

### DISCUSSION THREAD

**Discussion by frederikaalund** [empicano/aiomqtt#50] [General]
Jonathan Heathcote reached out to me to offer up the name of his [aiomqtt](https://github.com/mossblaser/aiomqtt) project. :+1:  It's a kind offer and I feel like we should accept it. :) If aiomqtt had been available at the time, I would have preferred it over asyncio-mqtt.

I'd like to hear your thoughts on the matter as well. Should we rename asyncio-mqtt to aiomqtt?

**Comment by flyte** [comment]
That's cool!

The client I started writing was called aio-paho-mqtt since it's still a wrapper around paho. In a perfect world I'd expect there  to be an aiohttp class pure-python MQTT client on aiomqtt but I don't currently know of any contenders.

I say take it on and maybe in the future if a super nice, stable, pure python client comes out of the woodwork then we could potentially pass on the mantle to them. Though of course the drawback is messing around the existing aiohttp users again with another different API. This may be mitigated somewhat by semver though.

**Reply by FlorianLudwig** [reply]
Shameless plug for https://github.com/Yakifo/aio-hbmqtt/issues/4 :)

With hbmqtt - a pure python mqtt implementation - being EOL'd by the author, we started a fork, looking for a new name... :]

**Reply by flyte** [reply]
Glad to see someone bringing hbmqtt up to date! If hbmqtt is being EOL'd then why not use "hbmqtt"? :grin:

**Reply by FlorianLudwig** [reply]
I would like that as well. But sadly, the author is unresponsive, https://github.com/beerfactory/hbmqtt/issues/235#issuecomment-775308640

**Reply by frederikaalund** [reply]
Thanks for the input. :+1: 

> The client I started writing was called aio-paho-mqtt since it's still a wrapper around paho.

I considered a similar name to begin with as well. :)) I was just a bit scared of using "paho" in the name since [Paho](https://www.eclipse.org/paho/) is an Eclipse thing. asyncio-mqtt is not sanctioned by either Eclipse or Paho. To avoid potential legal trouble, I chose a name without "paho" in it.

>  In a perfect world I'd expect there to be an aiohttp class pure-python MQTT client on aiomqtt but I don't currently know of any contenders.

There are [some non-paho-wrappers, asyncio-based alternatives](https://github.com/sbtinstruments/asyncio-mqtt#alternative-asyncio-based-mqtt-clients):

 * [hbmqtt](https://github.com/beerfactory/hbmqtt) is the most popular with 660 GitHub stars. [The API](https://hbmqtt.readthedocs.io/en/latest/references/mqttclient.html) leaves something to be desired IMHO. Certainly not what I would call idiomatic asyncio. I guess that's a matter of taste, though. Also, it's [officially unmaintained](https://github.com/beerfactory/hbmqtt/issues/235).
 * [distmqtt](https://github.com/M-o-a-T/distmqtt) is a fork of hbmqtt wtih 13 GitHub stars. [The API](https://distmqtt.readthedocs.io/en/latest/references/mqttclient.html) is much more idiomatic as well. Plus it's anyio-based (hint #44). The maintainer (smurfix) also maintains anyio, btw.
 * [gmqtt](https://github.com/wialon/gmqtt) with 189 GitHub stars. Again, I'd argue that the API is not that idiomatic. There is an [effort to improve the situation from](https://github.com/wialon/gmqtt/issues/69) November 2019. I proposed that they have a look at asyncio-mqtt for inspiration. Apparently that was too cheeky so the author marked my comment (and others) as disruptive. Oh well.

This list is not exhaustive; there are also a bunch of other libraries. The "asyncio-based MQTT libraries" space is really fragmented!

> I say take it on and maybe in the future if a super nice, stable, pure python client comes out of the woodwork then we could potentially pass on the mantle to them. 

I totally agree. I actually hope that happens at some point. :+1:

**Reply by frederikaalund** [reply]
Oh, I see that @FlorianLudwig got a reply in before me. Great to see that the work on hbmqtt continues as both aio-hbmqtt and distmqtt. :+1: Maybe you guys can merge your efforts? Maybe this is the library that deprecates asyncio-mqtt. :)
